economy/models.py

```diff
@@ -42,7 +48,61 @@ def invalid(self):
         return not self.valid
 
     def __str__(self):
-        return "Deposit for %s of %d NOK" % (self.person.name, self.amount)
+        return "Deposit by %s of %d NOK" % (self.person.username, self.amount)
 
     def __repr__(self):
-        return "Group(person=%s,amount=%d)" % (self.person.name, self.amount)
+        return "Group(person=%s,amount=%d)" % (self.person.username, self.amount)
+
+
+class Product(models.Model):
+    product_name = models.CharField(primary_key=True, max_length=50, blank=False, null=False, unique=True)
+    price = models.IntegerField(blank=False, null=False)
+
+    def __str__(self):
+        return "A product with name %s with the price of %d NOK" % (self.product_name, self.price)
+
+    def __repr__(self):
+        return "Product(product_name=%s, price=%d)" % (self.product_name, self.price)
+
+
+class PurchaseList(models.Model):
+    date_registered = models.DateField(auto_now_add=True, blank=False, null=False)
+    date_purchased = models.DateField(auto_now_add=True, blank=False, null=False)
+    signed_off_by = models.ForeignKey(User, blank=False, null=False)
+    comment = models.CharField(max_length=100, blank=False, null=False)
+
+    @property
+    def valid(self):
+        return self.signed_off_by is not None
+
+    @property
+    def invalid(self):
+        return not self.valid
+
+    def __str__(self):
+        return "A list of items purchased at date %s, registered at date %s by person %s" % \
+               (self.date_purchased, self.date_registered, self.signed_off_by.username)
+
+    def __repr__(self):
+        return "PurchaseList(date_purchased=%s, date_registered=%s, signed_off_by=%s, comment=%s)" % \
+               (self.date_purchased, self.date_registered, self.signed_off_by.username, self.comment)
+
+
+class Purchase(models.Model):
```

Comment:
The only thing I feel missing in this model is a `get_purchase_sum` method which calculates the total sum of the purchase, i.e. `product.price * amount`.

Not essential, but could come in handy.